doc/rst/language/evolution.rst

```diff
@@ -47,6 +47,125 @@ write.
+domain is modified. This overhead is unnecessary, for example, when the domain
+is operated upon by a single task.
+
+Motivated by this we have changed their default from
+``parSafe=true`` to ``parSafe=false``.
+With this change associative domains have been stabilized, except for domains
+requesting ``parSafe=true``.
+
+Here's a breakdown of the changes and how they might impact
+your programs:
+
+1. New default for associative domains:
+    * Previously, associative domains were ``parSafe`` by default. This has
+      changed to ``parSafe=false``. For example:
+
+      .. code-block:: chapel
+
+          var dom: domain(int);
+
+      used to imply that ``dom`` was set to ``parSafe=true`` but now it defaults
+      to ``parSafe=false``.
+    * This means that the checks to guarantee parallel safety are no longer
+      inserted by default, thus improving performance; furthermore, it now
+      requires additional diligence to ensure parallel safety.
+    * A warning will be generated for domains without an explicit ``parSafe``
+      setting unless you compile with ``-s noParSafeWarning``.
+
+      .. code-block:: chapel
+
+          var d1: domain(int);                 // warns
+          var d2: domain(int, parSafe=false);  // does not warn
+
+      where the compilation output of the above program would look as follows:
+
+      .. code-block:: console
+
+          $ chpl foo.chpl
+          foo.chpl:1: warning: The default parSafe mode for associative domains and arrays (like 'd1') is changing from 'true' to 'false'.
+          foo.chpl:1: note: To suppress this warning you can make your domain const, use an explicit parSafe argument (ex: domain(int, parSafe=false)), or compile with '-snoParSafeWarning'.
+          foo.chpl:1: note: To use the old default of parSafe=true, compile with '-sassocParSafeDefault=true'.
+
+    * Since ``const`` domains are never modified, they are exempt from these
+      warnings as changing their default to ``parSafe=false`` does not have the
+      potential to impact correctness.
+
+      .. code-block:: chapel
+
+          const dom: domain(int);  // does not warn
+
+    * In order to ease the transition, users can temporarily revert to the old
+      behavior by compiling with ``-s assocParSafeDefault=true``.
+
+      .. code-block:: console
+
+          $ chpl defaultAssociativeDomain.chpl -s assocParSafeDefault=true
+
+      If a program used associative domains and relied on ``parSafe=true``,
+      it might be useful to try compiling with ``-s assocParSafeDefault=true`` and
+      then add an explicit ``parSafe`` argument for each associative domain
+      individually, to ensure no races are introduced into the
+      program by forgoing the parallel safety guarantees.
+
+2. ``parSafe=true`` domains are unstable:
+    * Domains using ``parSafe=true`` are still considered unstable, and continue
+      to trigger unstable warnings when declared. For example:
```

Comment:
"and continue ..." --> ". Their declarations continue to ..."

Reply:
I think the single sentence flows better as it is. "Their declarations continue to trigger unstable warnings" sounds a bit staggered and break the continuum of the point being made.

Reply:
To me, an unstable warnings is triggered by a declaration, not by a domain. However I am OK with keeping the existing text, ideally removing the comma before 'and'.